clients/src/main/java/org/apache/kafka/common/telemetry/internals/MetricsEmitter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.internals;
+
+import java.util.Collections;
+import java.util.List;
+
+import java.io.Closeable;
+
+/**
+ * An {@code MetricsEmitter} emits the values held by the {@link SinglePointMetric}, likely first converting them
+ * to a format suitable for exposure, storage, or transmission. The telemetry reporter is likely
+ * the entity that is familiar with the underlying method of making the metrics visible to the
+ * broker. Thus, it is the primary place in the code where the implementation details are known.
+ *
+ * <p>
+ *
+ * An {@code MetricsEmitter} is stateless and the telemetry reporter should assume that the object is
+ * not thread safe and thus concurrent access to either the
+ * {@link #shouldEmitMetric(MetricKeyable)} or {@link #emitMetric(SinglePointMetric)} should be avoided.
+ *
+ * Regarding threading, the {@link #init()} and {@link #close()} methods may be called from
+ * different threads and so proper care should be taken by implementations of the
+ * {@code MetricsCollector} interface to be thread-safe. However, the telemetry reporter must
+ * ensure that the {@link #emitMetric(SinglePointMetric)} method should only be invoked in a synchronous
```

Comment:
I suppose this is true for all of the various emit methods.

Reply:
I think `shouldEmitMetric` is fine but have added `emittedMetrics` in the comments as well.